gateway/dataimpl/server_v1/queryadminserver.go

```diff
@@ -239,15 +330,39 @@ func (s *QueryIndexAdminServer) CreatePrimaryIndex(
 
 	_, err := s.executeQuery(ctx, &in.BucketName, qs)
 	if err != nil {
+		var name string
+		if in.Name == nil {
+			name = "#primary"
+		} else {
+			name = *in.Name
+		}
+
 		if errors.Is(err, cbqueryx.ErrIndexExists) {
-			var name string
-			if in.Name == nil {
-				name = "#primary"
-			} else {
-				name = *in.Name
-			}
 			return nil, s.errorHandler.NewQueryIndexExistsStatus(err, name).Err()
 		}
+
+		if errSt, ok := status.FromError(err); ok && errSt.Code() == codes.NotFound {
```

Comment:
Just so I understand here, how does this error get translated to codes.NotFound? Does query return a 404 or something?

Reply:
`executeQuery` calls `GetHttpOboAgent` that calls `getBucketAgent` which returns a `NewBucketMissingStatus` when the bucket named cannot be found which has the not found status.

Reply:
Sounds very reasonable to me, took me a while to figure out where the not found error was coming from when I was writing this.